pkg/util/arm/arm_test.go

```diff
@@ -115,6 +122,10 @@ type testResource struct {
 	Name        string                   `json:"name,omitempty"`
 	Unmarshaled int                      `json:"-"`
 	unexported  int
+	// Both `arm.Resource` and nested `testResource` have fields with name `Tags`.
+	// The `Tags` field from `arm.Resource` must override the one from `testResource`
+	// on the top-level of JSON.
+	Tags map[string]*string `json:"tags"`
```

Comment:
In the real Azure SDK structs almost all resources have similar `Tags` field.

Our custom `arm.Resource` also has the `Tags` field defined like this:

```go
Tags       map[string]interface{} `json:"tags,omitempty"`
```

Despite the fact that outer`Tags` field has `omitempty`, on a structure like this:

```go
&arm.Resource{
    Name:     "test",
    Resource: &testResource{},
}
```

we will get a json like this:

```json
{
  "name": "test",
  "tags": null
}
```

I believe `tags` should be omited in this case.

Reply:
**NOTE:** `"tags": null` was one of the reasons why vmss hashes were different in v3. I updated the implementation in https://github.com/openshift/openshift-azure/pull/2101 for consistency as well.